diff --git a/ffmpeg.c b/ffmpeg.c
index ad738cb..d4c0c93 100644
--- a/ffmpeg.c
+++ b/ffmpeg.c
@@ -1705,9 +1705,8 @@ static void forward_report(int is_last_report, int64_t timer_start, int64_t cur
         }
 
         // 5. calculate time
-        if (av_stream_get_end_pts(ost->st) != AV_NOPTS_VALUE)
-            pts = FFMAX(pts, av_rescale_q(av_stream_get_end_pts(ost->st),
-                                          ost->st->time_base, AV_TIME_BASE_Q));
+        if (ost->last_mux_dts != AV_NOPTS_VALUE)
+            pts = FFMAX(pts, ost->last_mux_dts);
 
         vid = 1;
     }
